src/lib/transaction_database/transaction_database.mli

```diff
@@ -0,0 +1,13 @@
+open Coda_base
+open Signature_lib
+
+type t
+
+val create : ?directory_name:string -> unit -> t
+
+val close : t -> unit
+
+val add : t -> Public_key.Compressed.t -> Transaction.t -> unit
+
+val get_transactions :
+  t -> Public_key.Compressed.t -> Transaction.t Non_empty_list.t option
```

Comment:
Is a `'a Non_empty_list.t option` better than a `'a list`?

Reply:
Yes, this is because when a query is successful, it would return at least one payment for a public key. If the database does not contain a certain public key, then it should return nothing rather than an empty list.

Reply:
`foo Non_empty_list.t option` is isomorphic to `foo list` though, and the latter is a little simpler.